safe/gui/tools/batch/batch_dialog.py

```diff
@@ -453,6 +617,55 @@ def write_report(self, report, pass_count, fail_count):
         except IOError:
             raise IOError
 
+    def generate_pdf_report(self, impact_function, iface, scenario_name):
+        """Generate map report and impact report from impact function and store
+        it in user specified directory. adapted from analysis utilites
+
+        :param impact_function: Impact Function
+        :param iface: iface
+        :return:
+        """
+        # output folder
+        output_dir = self.output_directory.text()
+        file_path = os.path.join(output_dir, scenario_name)
+
+        # create impact table report instance
+        table_report_metadata = ReportMetadata(
+            metadata_dict=standard_impact_report_metadata_pdf)
+        impact_table_report = ImpactReport(
+            iface,
+            table_report_metadata,
+            impact_function=impact_function)
+        impact_table_report.output_folder = file_path
+        impact_table_report.process_component()
+
+        # create impact map report instance
+        map_report_metadata = ReportMetadata(
+            metadata_dict=report_a4_blue)
+        impact_map_report = ImpactReport(
+            iface,
+            map_report_metadata,
+            impact_function=impact_function)
+        # Get other setting
+        settings = QSettings()
```

Comment:
this line and a few other lines below it have been commented out by Mas @lucernae in genereate_impact_map_report function in analysis_utilities.py, I am not sure on how to change it, but I think I will just follow the changes in that function because I haven't really check whether this function will also generate map pdf properly.